File: cmmc/schemas/datapact.py
"""Pydantic schemas for DataPact integration API."""

from datetime import datetime

from pydantic import BaseModel, Field


# ── Mapping schemas ──────────────────────────────────────────────────────────


class MappingCreate(BaseModel):
    practice_id: str = Field(..., min_length=1)
    datapact_contract_id: str = Field(..., min_length=1)
    datapact_contract_name: str | None = None


class MappingResponse(BaseModel):
    id: str
    org_id: str
    practice_id: str
    datapact_contract_id: str
    datapact_contract_name: str | None = None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class MappingListResponse(BaseModel):
    items: list[MappingResponse]
    total: int


# ── Sync schemas ─────────────────────────────────────────────────────────────


class SyncResultResponse(BaseModel):
    practice_id: str
    status: str
    message: str | None = None
    compliance: dict | None = None


class SyncResultsResponse(BaseModel):
    results: list[SyncResultResponse]


# ── Sync log schemas ─────────────────────────────────────────────────────────


class SyncLogResponse(BaseModel):
    id: str
    org_id: str
    assessment_id: str | None = None
    practice_id: str | None = None
    status: str
    error_message: str | None = None
    created_at: datetime

    model_config = {"from_attributes": True}


class SyncLogListResponse(BaseModel):
    items: list[SyncLogResponse]
    total: int


# ── Suggest schemas ──────────────────────────────────────────────────────────


class SuggestionResponse(BaseModel):
    practice_id: str
    contract_id: str
    contract_name: str | None = None
    reason: str | None = None

File: ui/src/types/datapact.ts
export interface Mapping {
  id: string
  org_id: string
  practice_id: string
  datapact_contract_id: string
  datapact_contract_name: string | null
  created_at: string
  updated_at: string
}

export interface MappingListResponse {
  items: Mapping[]
  total: number
}

export interface MappingCreate {
  practice_id: string
  datapact_contract_id: string
  datapact_contract_name?: string
}

export interface Contract {
  id: string
  title: string
  description?: string
  status?: string
  parties?: string[]
  created_at?: string
  updated_at?: string
}

export interface ContractListResponse {
  items: Contract[]
  total: number
}

export interface SyncResult {
  practice_id: string
  status: string
  message: string | null
  compliance: Record<string, unknown> | null
}

export interface SyncResultsResponse {
  results: SyncResult[]
}

export interface SyncLog {
  id: string
  org_id: string
  assessment_id: string | null
  practice_id: string | null
  status: string
  error_message: string | null
  created_at: string
}

export interface SyncLogListResponse {
  items: SyncLog[]
  total: number
}
